### PR TITLE
Add Order to the transaction

### PR DESCRIPTION
**Pull request by HoOngEe**

### CODE REVIEW

File: src/migrations/20190131075705-create-order.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+module.exports = {
+    up: (queryInterface, DataTypes) => {
+        return queryInterface.createTable("Orders", {
+            id: {
+                allowNull: false,
+                autoIncrement: true,
+                primaryKey: true,
+                type: DataTypes.BIGINT
+            },
+            orderHash: {
```
Comment (sgkim126) [inline src/migrations/20190131075705-create-order.js]: Isn't it `transactionHash`?

**Comment by sgkim126** [comment]
LGTM. @majecty How about you?

File: src/models/logic/orderontransfer.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    let orderOnTransferInstance: OrderOnTransferInstance;
+    try {
+        orderOnTransferInstance = await models.OrderOnTransfer.create({
+            transactionHash,
+            spentQuantity: orderOnTransfer.spentQuantity.value.toString(10),
+            inputIndices: orderOnTransfer.inputIndices,
+            outputIndices: orderOnTransfer.outputIndices
+        });
+
+        await createOrder(transactionHash, orderOnTransfer.order, networkId);
```
Comment (majecty) [inline src/models/logic/orderontransfer.ts]: Should we create a new order whenever the order is used in a transaction?
I know that an order could be used in many transactions, I worry that duplicate orders will be stored in the DB.

File: src/models/logic/transaction.ts [hunk 3]
```diff
@@ -430,6 +435,50 @@ async function handleUTXO(txInst: TransactionInstance, blockNumber: number) {
...
+                            order.lockScriptHashFrom
+                        );
+                        const parameters = order.parametersFrom;
+                        const quantity = new U64(order.assetQuantityFrom);
+                        const owner = getOwner(
+                            lockScriptHash,
+                            parameters,
+                            networkId
+                        );
+                        return createUTXO(
```
Comment (majecty) [inline src/models/logic/transaction.ts]: Could you explain to me about why we should create UTXOs from an order?
I think that the order only specifies which inputs/outputs are exchanged by this order. I don't know how a new UTXO could be created by order.

File: src/models/logic/orderontransfer.ts (hunk 2 above)
Comment (HoOngEe) [reply src/models/logic/orderontransfer.ts]: As you worried, it is problematic. This usage contradicts to the purpose of the introduction of OrderOnTransfer. I need more thinking then in what timing the order should be created.

File: src/models/logic/transaction.ts (hunk 3 above)
Comment (HoOngEe) [reply src/models/logic/transaction.ts]: I was wrong about UTXO. It seems no handling is need for orders, I'll remove above part.

File: src/models/logic/orderontransfer.ts (hunk 2 above)
Comment (HoOngEe) [reply src/models/logic/orderontransfer.ts]: Now the code check if order is already exists on DB.

**Pull request merged**